### ISSUE TITLE
exclusion of methods for next move and compiling with numba

### ISSUE THREAD

**Issue by Isabelle-Gbl** [eighthill/Slime_Mold_Simulation#29]
To improve the runtime of the code we should try to pass the coordinates of the agents into an array so that we can exclude the methods to update the next step of the agents from the class agent. That way we could compile it with numba.

**Comment by Isabelle-Gbl** [comment]
I have uploaded the code as far as I have come.
I have tried to extract all of the updating for the pheromones and agent moves out of the classes
that part is not done yet
also it is important that the code right now can only used within the simulation code and not within the GUI yet.

next steps would be:
- check if values are given correctly to the arrays
- check if calculations until line 107 are correct or if sth doesn't makes sense
- is there an easier way to write the code until 107?
- there are things that are not implemented yet: the max_value has to be found for each line in the array pheromone_values
   then those indizes can be used to find the correct colum for each line that has the values for updating the original agent_array
- then with this updated agent_array the pheromone array can be updated
